### PR TITLE
[SPARK-35523][DOCS] Fix the default value in Data Source Options page

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes fix the default value in Data Source Option page based on the Scala documentation.

### Why are the changes needed?

Some of the existing default value in Data Source Option page follow the Python documentation, which has `None` as the default value for all options.

### Does this PR introduce _any_ user-facing change?

Yes, the default value in the Data Source Option page is fixed (from `None` to proper default value)

- Before
Screen Shot 2021-06-02 at 6 31 12 PM

- After
Screen Shot 2021-06-02 at 6 32 47 PM

### How was this patch tested?

Manually built the docs and checked one by one.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
can we add default values at https://github.com/apache/spark/blob/5ff5770e5c4aeeec9c5f0ab173c49dfe003e5eba/docs/sql-data-sources-jdbc.md too?

File: docs/sql-data-sources-text.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ Data source options of text can be set via:
   </tr>
```
Comment (HyukjinKwon) [inline docs/sql-data-sources-text.md:57]: Can we change `wholetext`'s default value to `<code>false</code>`

File: docs/sql-data-sources-jdbc.md [hunk 2]
```diff
@@ -111,115 +120,144 @@ logging into the data sources.
...
     <td>
-        This is a JDBC writer related option. If enabled and supported by the JDBC database (PostgreSQL and Oracle at the moment), this options allows execution of a <code>TRUNCATE TABLE t CASCADE</code> (in the case of PostgreSQL a <code>TRUNCATE TABLE ONLY t CASCADE</code> is executed to prevent inadvertently truncating descendant tables). This will affect other tables, and thus should be used with care. This option applies only to writing. It defaults to the default cascading truncate behaviour of the JDBC database in question, specified in the <code>isCascadeTruncate</code> in each JDBCDialect.
+      This is a JDBC writer related option. If enabled and supported by the JDBC database (PostgreSQL and Oracle at the moment), this options allows execution of a <code>TRUNCATE TABLE t CASCADE</code> (in the case of PostgreSQL a <code>TRUNCATE TABLE ONLY t CASCADE</code> is executed to prevent inadvertently truncating descendant tables). This will affect other tables, and thus should be used with care.
     </td>
+    <td>write</td>
   </tr>
 
   <tr>
     <td><code>createTableOptions</code></td>
+    <td><code>""</code></td>
```
Comment (HyukjinKwon) [inline docs/sql-data-sources-jdbc.md:206]:
```suggestion
    <td><code></code></td>
```

**Comment by HyukjinKwon** [comment]
@itholic can you resolve conflicts?

**Review by HyukjinKwon — APPROVED** [review]

File: docs/sql-data-sources-json.md [hunk 3]
```diff
@@ -114,62 +114,62 @@ Data source options of JSON can be set via:
   <tr>
     <!-- TODO(SPARK-35433): Add timeZone to Data Source Option for CSV, too. -->
     <td><code>timeZone</code></td>
-    <td>None</td>
+    <td>The SQL config <code>spark.sql.session.timeZone</code></td>
```
Comment (HyukjinKwon) [inline docs/sql-data-sources-json.md:117]:
```suggestion
    <td>(value of <code>spark.sql.session.timeZone</code> configuration)</td>
```

Comment (HyukjinKwon) [reply docs/sql-data-sources-json.md:117]: to match with https://spark.apache.org/docs/latest/configuration.html

Comment (HyukjinKwon) [reply docs/sql-data-sources-json.md:117]: can you fix other instances too? e.g. in CSV as well https://github.com/apache/spark/blob/73fd6de9a18e8b550fd9afbcf9c87efa598fd76e/docs/sql-data-sources-csv.md

Comment (HyukjinKwon) [reply docs/sql-data-sources-json.md:117]: and parquet https://github.com/apache/spark/blob/73fd6de9a18e8b550fd9afbcf9c87efa598fd76e/docs/sql-data-sources-parquet.md

Comment (HyukjinKwon) [reply docs/sql-data-sources-json.md:117]: and avro https://github.com/apache/spark/blob/73fd6de9a18e8b550fd9afbcf9c87efa598fd76e/docs/sql-data-sources-avro.md

**Comment by HyukjinKwon** [comment]
can you also fix `ignoreNullFields` in JSON? it defaults to `spark.sql.jsonGenerator.ignoreNullFields`.

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by HyukjinKwon** [comment]
@itholic please keep the PR title with components properly next time

File: docs/sql-data-sources-jdbc.md [hunk 4]
```diff
@@ -43,34 +43,39 @@ following command:
...
+  * `DataFrameWriter`
 * `OPTIONS` clause at [CREATE TABLE USING DATA_SOURCE](sql-ref-syntax-ddl-create-table-datasource.html)
 
 For connection properties, users can specify the JDBC connection properties in the data source options.
 <code>user</code> and <code>password</code> are normally provided as connection properties for
 logging into the data sources.
 
 <table class="table">
-  <tr><th>Property Name</th><th>Meaning</th></tr>
+  <tr><th><b>Property Name</b></th><th><b>Default</b></th><th><b>Meaning</b></th><th><b>Scope</b></th></tr>
```
Comment (dongjoon-hyun) [inline docs/sql-data-sources-jdbc.md:55]: Do we need `<b>` inside `<th>`?

**Comment by dongjoon-hyun** [comment]
Thank you, @itholic and @HyukjinKwon .

File: docs/sql-data-sources-jdbc.md (hunk 4 above)
Comment (itholic) [reply docs/sql-data-sources-jdbc.md:55]: Actually I just followed the rule in existing files, and seems like It's a little bit more thicker 😅

- Before <b>

Screen Shot 2021-06-04 at 2 24 59 PM

- After <b>

Screen Shot 2021-06-04 at 2 25 34 PM

Comment (dongjoon-hyun) [reply docs/sql-data-sources-jdbc.md:55]: Thank you for checking! Looks better. 😄

**Pull request closed without merging**